include/swift/Basic/Statistic.h

```diff
@@ -13,7 +13,10 @@
 #ifndef SWIFT_BASIC_STATISTIC_H
 #define SWIFT_BASIC_STATISTIC_H
 
+#include "llvm/ADT/SmallString.h"
 #include "llvm/ADT/Statistic.h"
+#include "swift/Basic/Timer.h"
+#include "swift/FrontendTool/FrontendTool.h"
```

Comment:
I'm including FrontendTool.h for the definition of FrontendObserver (so that I can implement it), not the forward declarations of its parameters.

If you'd prefer, I could make the Basic/Statistic.h interface expose the full definitions of the  AlwaysOnFrontendCounters and AlwaysOnDriverCounters structs, and then implement the incrementing logic in each of Driver and Frontend (by implementing a special-purpose FrontendObserver subclass in FrontendTool itself, say). That would fix the layering violation, at least in the formal sense.

(Longer term I totally agree with you that I don't like all this information leakage between modules; but the LLVM global-static counters remain too expensive to be globally turned-on, and Matthias hasn't convinced LLVM folks of a suitable replacement strategy yet. We could implement our _own_ global-static registry system for counters, it just seemed to me like that would be a bit overkill.)